### PR TITLE
[GDI32_APITEST] Improve test for SetWorldTransform

### PR DESCRIPTION
**Pull request by tkreuzer**
Improve the test to check different types of invalid transformations and rounding errors.

### CODE REVIEW

**Comment by katahiromz** [comment]
Can you add the case of `(eM11, eM12, eM21, eM22) = (0, 1, 1, 0)`? Thanks.

**Comment by katahiromz** [comment]
You can add some tests for `GetWorldTransform` there.

File: modules/rostests/apitests/gdi32/SetWorldTransform.c [hunk 1]
```diff
@@ -29,15 +29,69 @@ void Test_SetWorldTransform()
...
+
+    /* Test world trnsform (should be unchanged by previous failure) */
+    result = GetWorldTransform(hdc, &xform);
+    ok(result == 1, "GetWorldTransform should succeed\n");
+    ok(xform.eM11 == 2, "xform.eM11 should be 2\n");
+    ok(xform.eM12 == (FLOAT)3.0001, "xform.eM11 should be 3.0001\n");
+    ok(xform.eM21 == 4, "xform.eM11 should be 4\n");
+    ok(xform.eM22 == 6, "xform.eM11 should be 6\n");
+
+    /* Set smalles possible values */
```
Comment (tkreuzer) [inline modules/rostests/apitests/gdi32/SetWorldTransform.c]:
```suggestion
    /* Set smallest possible values */
```

File: modules/rostests/apitests/gdi32/SetWorldTransform.c [hunk 2]
```diff
@@ -29,15 +29,69 @@ void Test_SetWorldTransform()
...
+    xform.eM12 = (FLOAT)0.9999999;
+    xform.eM21 = (FLOAT)1.0000001;
+    xform.eM22 = 1;
+    ok(xform.eM12 != (FLOAT)1.0, "xform.eM12 shouldn't be 1.0\n");
+    ok(xform.eM21 != (FLOAT)1.0, "xform.eM21 shouldn't be 1.0\n");
+    ok(xform.eM12 * xform.eM21 != (FLOAT)1.0, "xform.eM12 * xform.eM21 shouldn't be 1.0\n");
+    result = SetWorldTransform(hdc, &xform);
+    ok(result == 0, "SetWorldTransform should fail\n");
+
+    /* Test world trnsform (should be unchanged by previous failure) */
```
Comment (tkreuzer) [inline modules/rostests/apitests/gdi32/SetWorldTransform.c]:
```suggestion
    /* Test world transform (should be unchanged by previous failure) */
```

File: modules/rostests/apitests/gdi32/SetWorldTransform.c [hunk 3]
```diff
@@ -29,15 +29,69 @@ void Test_SetWorldTransform()
...
+    xform.eM22 = (FLOAT)1.4e-45;
+    ok(xform.eM11 != (FLOAT)0.0, "xform.eM11 shouldn't be 0.0\n");
+    ok(xform.eM22 != (FLOAT)0.0, "xform.eM22 shouldn't be 0.0\n");
+    ok(xform.eM11 * xform.eM22 != (FLOAT)0.0, "xform.eM12 * xform.eM21 shouldn't be 0.0\n");
+    result = SetWorldTransform(hdc, &xform);
+    ok(result == 1, "SetWorldTransform should succeed\n");
 
-	//pEntry = GdiHandleTable + GDI_HANDLE_GET_INDEX(hdc);
-	//pdcattr = pEntry->UserData;
+    /* Test world trnsform */
```
Comment (tkreuzer) [inline modules/rostests/apitests/gdi32/SetWorldTransform.c]:
```suggestion
    /* Test world transform */
```

**Pull request merged**